### PR TITLE
ENH: Don't infer WOM-5MON if we don't support it (#9425)

### LINKED ISSUE #9425: 'infer_freq' can return timerule (WOM-5MON) that 'to_offset' does not support
```python
import pandas as pd
index = [pd.Timestamp('2014-03-31 00:00:00'),
         pd.Timestamp('2014-06-30 00:00:00'),
         pd.Timestamp('2015-03-30 00:00:00'),
         pd.Timestamp('2015-06-29 00:00:00')]
freq = pd.infer_freq(index)  # WOM-5MON
pd.datetools.to_offset(freq)  # ValueError: Could not evaluate WOM-5MON
```

Instead of supporting WOM-5###, I think we can easily modify _get_wom_rule() to return None for this set of dates so it doesn't get inferred.  Agree?

### PR DESCRIPTION
**Pull request by craustin**
closes #9425

### CODE REVIEW

File: pandas/tseries/frequencies.py [hunk 1]
```diff
@@ -926,7 +926,8 @@ def _get_wom_rule(self):
             return None
 
         week_of_months = unique((self.index.day - 1) // 7)
-        if len(week_of_months) > 1:
+        week_of_months = week_of_months[week_of_months < 4]
```
Comment (jreback) [inline pandas/tseries/frequencies.py]: can you add a comment here on what this does and referene the issue (so future readers will know).

File: pandas/tseries/tests/test_frequencies.py [hunk 2]
```diff
@@ -210,6 +210,15 @@ def test_week_of_month(self):
         for day in days:
             for i in range(1, 5):
                 self._check_generated_range('1/1/2000', 'WOM-%d%s' % (i, day))
+            self._check_generated_range('1/1/2000', 'WOM-%d%s' % (i, day))
+
+    def test_fifth_week_of_month(self):
+        func = lambda: date_range('2014-01-01', freq='WOM-5MON')
```
Comment (jreback) [inline pandas/tseries/tests/test_frequencies.py]: add the issue reference as a comment here

**Comment by jreback** [comment]
pls add a release note. you can put in bug fixes for 0.17.0. looks good. pls ping when travis is green.

**Comment by craustin** [comment]
@jreback build passed. let me know if there's anything else i can do

**Comment by shoyer** [comment]
thanks!

**Pull request merged**